corehq/apps/consumer_user/views.py

```diff
@@ -0,0 +1,189 @@
+        ('token', AuthenticationTokenForm),
+        ('backup', BackupTokenForm),
+    )
+    invitation = None
+    signed_invitation_id = None
+    template_name = 'consumer_user/p_login.html'
+
+    @two_factor_exempt
+    @method_decorator(sensitive_post_parameters('password'))
+    def dispatch(self, request, *args, **kwargs):
+        if 'signed_invitation_id' in kwargs:
+            # User is using a link from an invitation
+            self.signed_invitation_id = kwargs['signed_invitation_id']
+            self.invitation = _get_invitation_or_400(self.signed_invitation_id)
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_form_kwargs(self, step=None):
+        """
+        Returns the keyword arguments for instantiating the form
+        (or formset) on the given step.
+        """
+        if self.invitation:
+            return {'invitation': self.invitation}
+        return {}
+
+    def get_success_url(self):
+        url = self.get_redirect_url()
+        if url:
+            return url
+        url = reverse('consumer_user:consumer_user_homepage')
+        return url
+
+    def get_context_data(self, **kwargs):
+        context = super(ConsumerUserLoginView, self).get_context_data(**kwargs)
+        if self.signed_invitation_id:
+            extra_context = {}
+            go_to_signup = reverse(
+                'consumer_user:consumer_user_register', kwargs={'signed_invitation_id': self.signed_invitation_id}
+            )
+            extra_context['go_to_signup'] = '%s%s' % (go_to_signup, '?create_user=1')
+            context.update(extra_context)
+        context.update({'hide_menu': True})
+        return context
+
+
+@two_factor_exempt
+@sensitive_post_parameters('password')
+def register_view(request, signed_invitation_id):
+
+    invitation = _get_invitation_or_400(signed_invitation_id)
+
+    if invitation.accepted:
+        return HttpResponseRedirect(reverse('consumer_user:consumer_user_login'))
+
+    email = invitation.email
+    create_user = request.GET.get('create_user', False)
+    if create_user != '1' and User.objects.filter(username=email).exists():
+        url = reverse(
+            'consumer_user:consumer_user_login_with_invitation',
+            kwargs={'signed_invitation_id': signed_invitation_id}
+        )
+        return HttpResponseRedirect(url)
+    if request.method == "POST":
+        form = ConsumerUserSignUpForm(request.POST, invitation=invitation)
+        if form.is_valid():
+            form.save()
+            return HttpResponseRedirect(reverse('consumer_user:consumer_user_login'))
+    else:
+        form = ConsumerUserSignUpForm(initial={'email': email})
+    return render(
+        request,
+        'consumer_user/signup.html',
+        {
+            'form': form,
+            'hide_menu': True,
+            'domain': invitation.domain,
+        }
+    )
+
+
+@consumer_user_login_required
+def success_view(request):
```

Comment:
Should this view be called `homepage`?  Looks like it's intended for use outside of just the initial signup workflow.